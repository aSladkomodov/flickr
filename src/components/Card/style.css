.card {
  flex: 1 0 300px;
  box-sizing: border-box;
  margin: 1rem 0.5em;

  &:hover {
    box-shadow: 1px 4px 16px 0px rgba(34, 60, 80, 0.2);
  }

  &-cover {
    height: 150px;

    > img {
      height: 100%;
      width: 100%;
    }
  }
}

@media screen and (min-width: 40em) {
  .card {
    max-width: calc(50% - 1em);
  }
}

@media screen and (min-width: 60em) {
  .card {
    max-width: calc(25% - 1em);
  }
}
