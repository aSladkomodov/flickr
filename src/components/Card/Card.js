import React from "react";
import CardWrapper from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

import "./style.css";

export const Card = ({ cover, date, description, tags }) => (
  <CardWrapper className="card">
    <CardHeader subheader={date} />
    <div className="card-cover">
      <img src={cover} alt="cover" loading="lazy" />
    </div>
    <CardContent>
      <Typography variant="body2" color="textSecondary" component="p">
        Description:
        <div
          dangerouslySetInnerHTML={{
            __html: description,
          }}
        />
      </Typography>
    </CardContent>
    <CardContent>
      <Typography variant="body2" color="textSecondary" component="p">
        Tags: {tags}
      </Typography>
    </CardContent>
  </CardWrapper>
);
