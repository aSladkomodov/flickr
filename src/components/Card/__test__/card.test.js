import React from "react";
import renderer from "react-test-renderer";
import { Card } from "../Card";

describe("Card component", () => {
  it("should render Card component", () => {
    const snapshot = renderer.create(
      <Card
        cover={"href"}
        title={"Title"}
        date={"08.09.2021"}
        description={"description"}
        tags={"tags"}
        link={"href"}
      />
    );
    expect(snapshot.toJSON()).toMatchSnapshot();
  });
});
