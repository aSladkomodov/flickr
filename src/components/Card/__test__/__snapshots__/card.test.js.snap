// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render Card component 1`] = `
<div
  className="MuiPaper-root MuiCard-root card MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        08.09.2021
      </span>
    </div>
  </div>
  <div
    className="card-cover"
  >
    <img
      alt="cover"
      loading="lazy"
      src="href"
    />
  </div>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Description:
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "description",
          }
        }
      />
    </p>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Tags: 
      tags
    </p>
  </div>
</div>
`;
